### DISCUSSION TITLE
JavaScript required?

### DISCUSSION THREAD

**Discussion by kevATin** [librecaptcha/lc-core#77] [Q&A] [answered]
Skipping over this project page I couldn't find the answer;
Is JavaScript required on the client to be able to solve the captchas?

And adding to that, is there any other requirement that would make these captchas not work in the Tor Browser on safest settings?

**Comment by hrj** [comment, marked as answer]
This project (what we call the core) doesn't provide any frontend. It is just a backend service that provides an API. Your app will need to use the APIs to create a user facing website, with or without javascript.

The current CAPTCHA samples are simple image-with-text type of CAPTCHAs, so it should be possible to use them with no javascript at all.

We do want to add other types of CAPTCHA samples, which might require some interactivity and hence JS. But it will be always possible to turn them off through the configuration file.

If you are seriously planning to use this, please wait till our next release, before deploying in production. We are trying to address these [issues](https://github.com/librecaptcha/lc-core/milestone/4)

**Reply by kevATin** [reply]
I see, it's just very important to me that no one is locked out of using this, as is the case with reCaptcha. 
I wonder if it would be feasible to do audio captchas without javascript.

**Comment by hrj** [comment]
Also, we do want to create a few sample frontends in the future. Which backend framework are you planning to use?

**Reply by kevATin** [reply]
I have no solid plans for anything, the main reason I was asking is because I know a lot of (mostly open source related) sites are looking for captcha solutions, and not requiring Javascript/being Tor-compatible would be a huge upside.

**Comment by hrj** [comment]
> And adding to that, is there any other requirement that would make these captchas not work in the Tor Browser on safest settings?

(Ooops, almost missed this)

The answer is: it will be upto the app to ensure this.

PS. FWIW, I am developing [a browser](https://github.com/gngrOrg/gngr) with JS off by default 😄
